dbms/src/Common/UInt128.h

```diff
@@ -155,10 +176,26 @@ struct UInt256HashCRC32
 };
 
 #endif
+}
 
+/// Overload hash for type casting
```

Comment:
Inside the `ToIntMonotonicity` on `Functions/FunctionsConversion.h`, they instantiate the `DataTypeEnum` class who have a `std::unordered_map` who have the template type `ToIntMonotonicity` as a key. (so for example who interest us, if we instantiate `ToIntMonotonicity` with an UInt128, it will instantiate a `std::unordered_map` who will have a UInt128 as a key, and `std::unordered_map` need to have the `operator()()` available)

So when I talk about type casting, it's most the conversion functions. (`toInt8`, `toUInt32`, etc...)